src/udev/udevd.c

```diff
@@ -348,7 +348,7 @@ static int worker_lock_block_device(sd_device *dev, int *ret_fd) {
         if (r < 0)
                 return log_device_debug_errno(dev, r, "Failed to get the value of property 'ACTION': %m");
 
-        if (streq(val, "remove"))
+        if (STR_IN_SET(val, "remove", "unbind"))
```

Comment:
No. Since the device still exists, something might be massaging it, and we want to serialize access. I imagine the following scenario: a driver is unbound to start a firmware upgrade. Once unbound, firmware upload is started. After upload, the driver is bound again, and the device can be used. All steps should be serialized wrt. to udevd and other things that look at the device.